### PR TITLE
object deep assign

### PR DESCRIPTION
**Pull request by LEEDASILVA**

### CODE REVIEW

File: exercise/replicate.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+const $a = {
+  arr: [23, 1, 34, 5, 5],
+  obj: { something: 'hello', arr2: [1, 0] },
+}
+
+const $f1 = { a: { b: 3 } }
+const $f2 = { a: { b: 3, c: { d: { e: 'hello' } } } }
+const $f3 = { a: { b: 4, c: { d: { e: 'xau' } } } }
+const $f = { a: { b: 4, c: { d: { e: 'xau' } } } }
```
Comment (kigiri) [inline exercise/replicate.js]: we could add a quote from blade runner:
> Replicants weren't supposed to have feelings... neither were blade runners.
>
> - Rick Deckard

File: tests/replicate/fail_JSON.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+const replica = (target, ...obj) =>
+  JSON.parse(JSON.stringify(Object.assign(target, ...obj)))
```
Comment (kigiri) [inline tests/replicate/fail_JSON.js]: I don't think that this tests properly that we didn't use `JSON` to copy as this is still a shalow copy

File: tests/replicate/pass_ms.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const replica = (target, ...obj) => copyObject(Object.assign(target, ...obj))
```
Comment (kigiri) [inline tests/replicate/pass_ms.js]: I'm pretty sure I can write a test that makes this fail, i'll give it a shot

**Review by kigiri — CHANGES_REQUESTED** [review]
So we talk about it on discord:
- Add more tests
- Make tests more descriptive to what is being tested
- Split in 2 exercises `assignDeep` and `copyDeep`

File: exercise/replicate.js (hunk 1 above)
Comment (LEEDASILVA) [reply exercise/replicate.js]: ohh ok cool

File: tests/replicate/fail_JSON.js (hunk 2 above)
Comment (LEEDASILVA) [reply tests/replicate/fail_JSON.js]: i did a separated pr with the deep copy and the fail test https://github.com/nan-academy/js-training/pull/70

File: tests/replicate/pass_ms.js (hunk 3 above)
Comment (LEEDASILVA) [reply tests/replicate/pass_ms.js]: yup you are right it fails. I corrected the solution and the subject. But i will update my solution

**Review by kigiri — APPROVED** [review]

**Pull request merged**